### ISSUE TITLE
bug: Run the CLI 'dev' command Error - "Ngrok failed to create a tunnel"

### ISSUE THREAD

**Issue by birksy89** [triggerdotdev/trigger.dev#488]
### Provide environment information

  System:
    OS: Windows 10 10.0.19045
    CPU: (8) x64 Intel(R) Core(TM) i7-6700 CPU 
@ 3.40GHz
    Memory: 6.00 GB / 15.95 GB
  Binaries:
    Node: 18.16.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.18 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 8.4.0 - ~\AppData\Roaming\npm\npm.CMD 
    pnpm: 7.13.4 - ~\AppData\Roaming\npm\pnpm.CMD

### Describe the bug

![image](https://github.com/triggerdotdev/trigger.dev/assets/4971630/d5336a6d-cac5-4b5a-ad09-5acf0249b147)

![image](https://github.com/triggerdotdev/trigger.dev/assets/4971630/bdb5db20-a5b6-4c7b-961f-ef9e92f6ec5b)

### Reproduction repo

N/A

### To reproduce

```
npx @trigger.dev/cli@latest dev
```

### Additional information

I do have ngrok installed already, from previous projects

**Comment by birksy89** [comment]
No, I'm not running ngrok already.

I can run:
```
ngrok http 3000
```

And this works successfully (Doesn't say port 3000 is already in use etc)

I can then terminate than, and try to run:
```
npx @trigger.dev/cli@latest dev
```

Which by default runs on port 3000, and it fails.

I can then attempt to run:
```
next dev -p 8080
```

To switch the port to 8080, and attempt to run:
```
npx @trigger.dev/cli@latest dev
```
But that also fails:
![image](https://github.com/triggerdotdev/trigger.dev/assets/4971630/98092853-d2a6-429b-81fe-c30fd97e1452)

**Comment by Chigala** [comment]
Hello @birksy89, I apologize for the delayed response. I have been unable to reproduce this with the same error you encountered.  Although, if `ngrok http 3000` works, then this should work too, under the hood they both  perform the same operation.

@ericallam @matt-aitken  What do you think?

**Comment by birksy89** [comment]
I found this line:
https://github.com/triggerdotdev/trigger.dev/blob/main/packages/cli/src/commands/dev.ts#L296
```
error.message.includes("connect ECONNREFUSED 127.0.0.1:4041")
```

As you can see from the screenshot, mine is saying the same, but for port `4040`.

I restarted my machine, and can't see ngrok running, unless I explicitly open a terminal and run the ngrok command.

**Comment by birksy89** [comment]
It also doesn't work for me when attempting to follow the "fresh install" onboarding pathway:

![image](https://github.com/triggerdotdev/trigger.dev/assets/4971630/5305d487-c063-4d81-a426-d306f333fd16)

**Comment by Chigala** [comment]
> I found this line: https://github.com/triggerdotdev/trigger.dev/blob/main/packages/cli/src/commands/dev.ts#L296
> 
> ```
> error.message.includes("connect ECONNREFUSED 127.0.0.1:4041")
> ```
> 
> As you can see from the screenshot, mine is saying the same, but for port `4040`.
> 
> I restarted my machine, and can't see ngrok running, unless I explicitly open a terminal and run the ngrok command.

I implemented the quoted line of code in response to a user's complaint about being unable to establish a tunnel while having another instance of ngrok running. This code was added to provide a clear error message explaining why it is not possible to create a tunnel in such a situation. 

In your case ngrok cannot connect to port 4040 which it is usually pointed to by default. 
- Run this command on your terminal to check if any application is currently running on port 4040, I see you are using windows machine. 
` netstat -an | find "LISTENING `
-  if another application is running on the port, you can stop the application or run this command, `$ ngrok config check`, this would show the location of your ngrok config file where you can then change the default port by adding this . `web_addr: 4045`. 
- Now you can run `npx @trigger.dev/cli@latest dev `  again and hopeful this would fix it. 

If that doesn't fix it and also there is nothing running on port 4040 on your machine, maybe you can update your ngrok to the latest version and try. As of writing this,  I currently use ngrok version 3.3.2. 

I really hope this helps

**Comment by birksy89** [comment]
Hello @Chigala,

I've managed to get it working. Here are the steps I took:

I ran those commands, and confirmed that nothing was showing up on port 4040

I attempted to upgrade ngrok via it's `ngrok update` command, but was told I was on the highest version - Even though I wasn't. I was on a v2, and I know there's a v3.

The way I had ngrok installed was as a node-wrapper / npm package - So I did `npm un ngrok -g`

I then tried to use `npx @trigger.dev/cli@latest dev`, but it didn't work still.

I then went to modify the config file which exists here for me:
`C:\Users\Andrew\AppData\Local\ngrok\ngrok.yml`

Altering the `web_addr: 4045` yielded a different error:
![image](https://github.com/triggerdotdev/trigger.dev/assets/4971630/d81fd3a2-62f2-477d-98b3-0b1284301449)

I deleted the file, and ran a fresh config command:
`ngrok config add-authtoken new-auth-token-value-from-ngrok-dashboard`

This span up a new yml file, and uses the 4040 port by default still, I assume.

And then it worked 🎉 

So in short, the issue was the value of the `authtoken` in the `ngrok.yml` file
